test/unit/org/apache/cassandra/tools/JMXStandardsTest.java

```diff
@@ -0,0 +1,180 @@
+                                                 .add(ExecutionException.class)
+                                                 .add(InterruptedException.class)
+                                                 .add(UnknownHostException.class)
+                                                 .add(IOException.class)
+                                                 .add(TimeoutException.class)
+                                                 .add(IllegalStateException.class)
+                                                 .add(ClassNotFoundException.class)
+                                                 .add(OpenDataException.class)
+                                                 .build();
+    /**
+     * This list is a set of types under java.* and javax.*, but are too vague that could cause issues; this does not
+     * mean issues will happen with JMX, only that issues may happen only after running and can not be detected at
+     * compile time.
+     */
+    private static Set<Class<?>> DANGEROUS_TYPES = ImmutableSet.<Class<?>>builder()
+                                                   .add(Object.class)
+                                                   .add(Comparable.class)
+                                                   .add(Serializable.class)
+                                                   .add(Exception.class)
+                                                   .build();
+    @Test
+    public void interfaces() throws ClassNotFoundException
+    {
+        Reflections reflections = new Reflections(ConfigurationBuilder.build("org.apache.cassandra").setExpandSuperTypes(false));
+        Pattern mbeanPattern = Pattern.compile(".*MBean$");
+        Set<String> matches = reflections.getAll(Scanners.SubTypes).stream()
+                                         .filter(s -> mbeanPattern.matcher(s).find())
+                                         .collect(Collectors.toSet());
+
+        List<String> warnings = new ArrayList<>();
+        List<String> errors = new ArrayList<>();
+        for (String className : matches)
+        {
+            for (Class<?> klass = Class.forName(className); klass != null && !Object.class.equals(klass); klass = klass.getSuperclass())
+            {
+                Assertions.assertThat(klass).isInterface();
+                Method[] methods = klass.getDeclaredMethods();
+                for (int i = 0; i < methods.length; i++)
+                {
+                    Method method = methods[i];
+                    checkType(method, "return", method.getGenericReturnType(), warnings, errors);
+                    Stream.of(method.getGenericParameterTypes()).forEach(t -> checkType(method, "parameter", t, warnings, errors));
+                    Stream.of(method.getGenericExceptionTypes()).forEach(t -> checkType(method, "throws", t, warnings, errors));
+                }
+            }
+        }
+        if (!warnings.isEmpty())
+            System.err.println(String.join("\n", warnings));
```

Comment:
Why don't we produce multiple error messages? Shouldn't we emit log warnings instead?

Reply:
Replacing with

```
warnings.forEach(logger::warn);
```